arabot/modules/games.py

```diff
@@ -380,50 +379,81 @@ def check_board_winner(self) -> disnake.abc.User | bool:
         return all(cell for row in self.board for cell in row)
 
 
+class RussianRoulette:
+    GAMES: ClassVar[dict[int, Self]] = {}
+
+    def __init__(self, chambers: int = 6, mega_bullet_deaths_threshold: int = 3):
+        self.current_pos = 0
+        self.chambers = chambers
+        self.bullet_pos = random.randint(1, chambers)
+        self.last_shooter_id = 0
+        self.last_deaths = deque[int](maxlen=mega_bullet_deaths_threshold)
+
+    def __repr__(self) -> str:
+        return f"{__class__.__name__}({self.current_pos}, {self.bullet_pos})"
+
+    @classmethod
+    def get_game(cls, guild_id: int) -> Self:
+        if rev := cls.GAMES.get(guild_id):
+            return rev
+        rev = cls.GAMES[guild_id] = cls()
+        return rev
+
+    def shoot(self, user_id: int) -> bool:
+        self.last_shooter_id = user_id
+        self.current_pos += 1
+        if kill := self.current_pos == self.bullet_pos:
+            self.last_deaths.append(user_id)
+        return kill
+
+    def killed(self) -> bool:
+        return self.current_pos >= self.bullet_pos
+
+    def mega_killed(self) -> bool:
+        return self.last_deaths.count(self.last_shooter_id) == self.last_deaths.maxlen
+
+    def reload(self) -> None:
+        if self.mega_killed():
+            self.last_deaths.clear()
+        self.current_pos = 0
+        self.bullet_pos = random.randint(1, self.chambers)
+        self.last_shooter_id = 0
+
+
+def rr_cooldown(msg: disnake.Message) -> commands.Cooldown | None:
+    rev = RussianRoulette.GAMES.get(msg.guild.id)
+    if rev and rev.last_shooter_id != msg.author.id and rev.killed():
```

Comment:
You always run RussianRoulette.reload()
before 
RussianRoulette.killed()
this means statement will be forever false